common/src/main/java/org/apache/celeborn/common/network/client/TransportClientFactory.java

```diff
@@ -162,6 +169,9 @@ public TransportClient createClient(String remoteHost, int remotePort, int parti
         if (cachedClient.isActive()) {
           logger.trace("Returning cached connection to {}: {}", resolvedAddress, cachedClient);
           return cachedClient;
+        } else if (cachedClient.getChannel().isActive()) {
+          logger.info("Found timeout connection to {}, creating a new one.", cachedClient.getSocketAddress());
+          danglingChannels.add(cachedClient.getChannel());
```

Comment:
Seems danglingChannels will only be cleared in close() which is possibly never called, so danglingChannels may grow infinitely. Do we really need it?

Reply:
danglingChannels is added to make it backward compatible. otherwise, if channel is directly closed for heartbeat timeout, inflight request that takes time longer than heartbeat interval * wait times may be closed.
in short, it may conflict with the original io.connectionTimeout config.

Reply:
alse it wont grow infinitely, as io.connectionTimeout config will close idle connections at last